src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -248,6 +248,49 @@ const exprt java_bytecode_convert_methodt::variable(
   }
 }
 
+// In order to construct the method type, we can either use signature or
+// descriptor. Since only signature contains the generics info, we want to
+// use signature whenever possible. We revert to using descriptor if (1) the
+// signature does not exist, (2) an unsupported generics are present or
+// (3) in the special case when the number of parameters in the descriptor does
+// not match the number of parameters in the signature - e.g. for certain
+// types of inner classes and enums (see unit tests for examples).
+typet member_type_lazy(const std::string &descriptor,
+                       const optionalt<std::string> &signature,
+                       const std::string &class_name,
+                       message_handlert &message_handler)
```

Comment:
I think you can make the `message_handler` parameter be `const`?

Reply:
Ah, as @svorenova pointed out to me, `message_handler` can't actually be `const`, so this comment is garbage...